service/src/main/java/com/ericsson/eiffel/remrem/generate/controller/RemremGenerateController.java

```diff
@@ -36,16 +39,30 @@
     * 
     */
     @RequestMapping(value = "/{mp}", method = RequestMethod.POST)
-    public JsonElement generate(@PathVariable String mp, @RequestParam("msgType") String msgType,
+    public ResponseEntity<?> generate(@PathVariable String mp, @RequestParam("msgType") String msgType,
             @RequestBody JsonObject bodyJson) {
         MsgService msgService = getMessageService(mp);
-        if (msgService != null) {
-            return parser.parse(msgService.generateMsg(msgType, bodyJson));
-        } else {
-            return null;
+        String response= "";
+        try{
+            if (msgService != null) {
+            	response = msgService.generateMsg(msgType, bodyJson);
+                if(!response.contains("message")) {
+                	return new ResponseEntity<>(parser.parse(response),HttpStatus.OK);
+                }
+                else {
+                	return new ResponseEntity<>(parser.parse(response),HttpStatus.BAD_REQUEST);
+                }
+            }
+            else {
+            	return new ResponseEntity<>(parser.parse(RemremGenerateServiceConstants.NO_SERVICE_ERROR),HttpStatus.SERVICE_UNAVAILABLE);
```

Comment:
Handle this in the Message Service and get the Error Message as we get it for HttpStatus.BAD_REQUEST